weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts

```diff
@@ -202,12 +203,18 @@ const useCalls = (
   filter: CallFilter,
   limit?: number,
   opts?: {skip?: boolean}
-): Loadable<CallSchema[]> => {
+): LoadableWithRefetch<CallSchema[]> => {
   const getTsClient = useGetTraceServerClientContext();
   const loadingRef = useRef(false);
   const [callRes, setCallRes] =
     useState<traceServerClient.TraceCallsQueryRes | null>(null);
   const deepFilter = useDeepMemo(filter);
+
+  const [doRefetch, setDoRefetch] = useState(false);
+  const refetch = useCallback(() => {
+    setDoRefetch(r => !r);
+  }, []);
```

Comment:
this is a hack!

Reply:
while this is a hack, you have the right idea. The implementation is a bit odd. Perhaps consider having the state be a timestamp, and refreshing sets the timestamp to the current timestamp.

Reply:
In the future, we will have polling and better mechanics here